lib/commands/app-management.js

```diff
@@ -1,176 +1,497 @@
+    '-f',
+    '0x10200000',
+    '-n',
+    activityName,
+  ]);
+  this.log.debug(stdout);
+  if (/^error:/im.test(stdout)) {
+    throw new Error(`Cannot activate '${appId}'. Original error: ${stdout}`);
+  }
+}
+
+/**
+ * @this {import('../driver').AndroidDriver}
+ * @param {import('./types').ActivateAppOpts} opts
+ * @returns {Promise<void>}
+ */
+export async function mobileActivateApp(opts) {
+  const {appId} = requireArgs('appId', opts);
+  return await this.activateApp(appId);
+}
+
+/**
+ * @this {import('../driver').AndroidDriver}
+ * @param {string} appId
+ * @param {Omit<import('appium-adb').UninstallOptions, 'appId'>} opts
+ * @returns {Promise<boolean>}
+ */
+export async function removeApp(appId, opts = {}) {
+  return await this.adb.uninstallApk(appId, opts);
+}
+
+/**
+ * @this {import('../driver').AndroidDriver}
+ * @param {import('./types').RemoveAppOpts} opts
+ * @returns {Promise<boolean>}
+ */
+export async function mobileRemoveApp(opts) {
+  const {appId} = requireArgs('appId', opts);
+  return await this.removeApp(appId, opts);
+}
+
+/**
+ * @this {import('../driver').AndroidDriver}
+ * @param {string} appId
+ * @param {Omit<import('./types').TerminateAppOpts, 'appId'>} [options={}]
+ * @returns {Promise<boolean>}
+ */
+export async function terminateApp(appId, options = {}) {
+  this.log.info(`Terminating '${appId}'`);
+  if (!(await this.adb.processExists(appId))) {
+    this.log.info(`The app '${appId}' is not running`);
+    return false;
+  }
+  await this.adb.forceStop(appId);
+  const timeout =
+    util.hasValue(options.timeout) && !Number.isNaN(options.timeout)
+      ? parseInt(String(options.timeout), 10)
+      : 500;
+
+  if (timeout <= 0) {
+    this.log.info(
+      `'${appId}' has been terminated. Skip checking the application process state ` +
+        `since the timeout was set as ${timeout}ms`,
+    );
+    return true;
+  }
+
+  try {
+    await waitForCondition(async () => (await this.queryAppState(appId)) <= APP_STATE.NOT_RUNNING, {
+      waitMs: timeout,
+      intervalMs: 100,
+    });
+  } catch (e) {
+    this.log.errorAndThrow(`'${appId}' is still running after ${timeout}ms timeout`);
+  }
+  this.log.info(`'${appId}' has been successfully terminated`);
+  return true;
+}
+
+/**
+ * @this {import('../driver').AndroidDriver}
+ * @param {import('./types').TerminateAppOpts} opts
+ * @returns {Promise<boolean>}
+ */
+export async function mobileTerminateApp(opts) {
+  const {appId} = requireArgs('appId', opts);
+  return await this.terminateApp(appId, opts);
+}
+
+/**
+ * @this {import('../driver').AndroidDriver}
+ * @param {string} appPath
+ * @param {Omit<import('appium-adb').InstallOptions, 'appId'>} opts
+ * @returns {Promise<void>}
+ */
+export async function installApp(appPath, opts) {
+  const localPath = await this.helpers.configureApp(appPath, APP_EXTENSIONS);
+  await this.adb.install(localPath, opts);
+}
+
+/**
+ * @this {import('../driver').AndroidDriver}
```

Comment:
with so many `import`s it might be nice to just define the `AndroidDriver` type once below and refer to it elsewhere

Reply:
done